bpfctl/src/main.rs

```diff
@@ -158,6 +160,235 @@ struct GlobalArg {
+                let proc_on = match TcProceedOn::from_strings(proceed_on) {
+                    Ok(p) => p,
+                    Err(e) => bail!("error parsing proceed_on {e}"),
+                };
+                Ok(Some(load_request::AttachInfo::TcAttachInfo(TcAttachInfo {
+                    priority: *priority,
+                    iface: iface.to_string(),
+                    position: 0,
+                    direction: direction.to_string(),
+                    proceed_on: proc_on.as_action_vec(),
+                })))
+            }
+            LoadCommands::Tracepoint { tracepoint } => Ok(Some(
+                load_request::AttachInfo::TracepointAttachInfo(TracepointAttachInfo {
+                    tracepoint: tracepoint.to_string(),
+                }),
+            )),
+        }
+    }
+}
+
+impl Commands {
+    fn get_request_common(&self) -> anyhow::Result<Option<LoadRequestCommon>> {
+        let id: &Option<String>;
+        let section_name: &String;
+        let global: &Option<Vec<GlobalArg>>;
+        let command: &LoadCommands;
+        let location: Option<load_request_common::Location>;
+
+        let mut global_data: HashMap<String, Vec<u8>> = HashMap::new();
+
+        // Common fields
+        match self {
+            Commands::LoadFromFile(l) => {
+                id = &l.id;
+                section_name = &l.section_name;
+                global = &l.global;
+                command = &l.command;
+                location = Some(load_request_common::Location::File(l.path.clone()));
+            }
+            Commands::LoadFromImage(l) => {
+                id = &l.id;
+                section_name = &l.section_name;
+                global = &l.global;
+                command = &l.command;
+                location = {
+                    let image_pull_policy: ImagePullPolicy = l
+                        .image_pull_policy
+                        .as_str()
+                        .try_into()
+                        .expect("invalid image pull policy");
+                    let (username, password) = match l.registry_auth.clone() {
+                        Some(a) => {
+                            let auth_raw = general_purpose::STANDARD_NO_PAD.decode(a)?;
+                            let auth_string = String::from_utf8(auth_raw)?;
+                            let (username, password) = auth_string.split(':').next_tuple().unwrap();
+
+                            (username.to_owned(), password.to_owned())
+                        }
+                        None => ("".to_owned(), "".to_owned()),
+                    };
+
+                    Some(load_request_common::Location::Image(BytecodeImage {
+                        url: l.image_url.clone(),
+                        image_pull_policy: image_pull_policy as i32,
+                        username,
+                        password,
+                    }))
+                };
+            }
+            _ => bail!("Unknown command"),
+        };
+
```

Comment:
```suggestion
        // Common fields
        let common = match self {
            Commands::LoadFromFile(l) => {
                LoadRequestCommon{
                    id: &l.id;
                    section_name: &l.section_name;
                    global: &l.global;
                    command: &l.command;
                    location: Some(load_request_common::Location::File(l.path.clone()));
                }
            }
            Commands::LoadFromImage(l) => {
                LoadRequestCommon{
                    id: &l.id;
                    section_name: &l.section_name;
                    global: &l.global;
                    command: &l.command;
                    location: {
                        let image_pull_policy: ImagePullPolicy = l
                            .image_pull_policy
                            .as_str()
                            .try_into()
                            .expect("invalid image pull policy");
                        let (username, password) = match l.registry_auth.clone() {
                            Some(a) => {
                                let auth_raw = general_purpose::STANDARD_NO_PAD.decode(a)?;
                                let auth_string = String::from_utf8(auth_raw)?;
                                let (username, password) = auth_string.split(':').next_tuple().unwrap();

                                (username.to_owned(), password.to_owned())
                            }
                            None => ("".to_owned(), "".to_owned()),
                        };

                        Some(load_request_common::Location::Image(BytecodeImage {
                            url: l.image_url.clone(),
                            image_pull_policy: image_pull_policy as i32,
                            username,
                            password,
                        }))
                    };
                }
            }
            _ => bail!("Unknown command"),
        };

```

Reply:
Hmmm, I would not say I'm fan of the proposal (if I understand it correctly). I pushed an updated version. There is now code duplication for common fields (.to_string() ...) and still fetching `global` from different types require a variable.